app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java

```diff
@@ -100,38 +89,24 @@ protected String getTrackedViewName() {
         return null;
     }
 
+    @NonNull
     public GoogleApiClient getGoogleApiClient() {
-        return mGoogleApiClient;
+        return App.getInstance().getGoogleApiClient();
     }
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
-        if (!Utils.isEmulator()) {
-
-            mGoogleApiClient = new GoogleApiClient.Builder(this)
-                    .addConnectionCallbacks(this)
-                    .addOnConnectionFailedListener(this)
-                    .addApi(Plus.API)
-                    .addApi(Games.API)
-                    .addApi(AppStateManager.API)
-                    .addScope(Plus.SCOPE_PLUS_LOGIN)
-                    .addScope(Plus.SCOPE_PLUS_PROFILE)
-                    .addScope(Games.SCOPE_GAMES)
-                    .build();
-        }
-
-        mAchievementActionHandler =
-                new AchievementActionHandler(getHandler(), mGoogleApiClient, this);
+        getGoogleApiClient().registerConnectionFailedListener(this);
```

Comment:
Do you know what is the difference between doin it like this or using `addConnectionCallbacks` in the builder. When we do it in the builder we never unregister.